frontend/src/diagram/DiagramWebSocketContainer.types.ts

```diff
@@ -58,6 +58,8 @@ export interface CreateEdgeTool extends Tool {
   edgeCandidates: EdgeCandidate[];
 }
 
+export interface DeleteTool extends Tool {}
+
```

Comment:
You are not describing the content of the DeleteTool here, yet you are manipulating its properties later in the code with code like `tool.targetDescriptions`. It would be better if you were really using TypeScript because I suspect that this type that you have introduced is almost unused since it does not match the reality.

Reply:
You're not the first one to write such code in the diagram parts but it would be better if it was done incrementally. I won't block the PR for that but keep that in mind next time since one day, I will turn on the `noImplicitAny` setting in our TypeScript configuration.